### DISCUSSION TITLE
Pre bug report discussion: Empty transfers

### DISCUSSION THREAD

**Discussion by chrispink** [iterate-ch/cyberduck#14902] [Q&A]
MacOS 12.6.3 Cyberduck 8.6.0

I am getting persistent empty transfers (0 B). As far as I can tell, it's only with overwrites.

The Transfer window reports successful, with a summary of bytes transferred but the file is empty and the listing shows 0 B
Repeating the transfer give the same result
The only way to work around this behaviour is to quit and restart Cyberduck. This consistently resolves the issue temporarily.

This is happening since the previous version of Cyberduck, the version above was an update to try and resolve the issue.

Does anyone else have this issue?

**Comment by tatze96** [comment]
Is a specific protocol (e.g. WebDAV, S3, OneDrive, etc.) affected?
Does the transfer behavior occur in combination with specific files (file extension, file size)?

**Reply by chrispink** [reply]
It's standard SFTP to  cPanel accounts on Port 22. 
I'm trying to establish whether it's one of our servers that's affected or all. 

The file type seems arbitrary and the size is generally small. The main characteristic is that the Transfer window shows x KB of x KB transferred with a green bullet icon whereas the file listing shows immediately 0 B

**Reply by chrispink** [reply]
More information: It's a cPanel server with "Cookie IP validation" set to strict.

Cyberduck says it's making a new connection each time it starts a transfer but, I'm guessing, it used the same IP address for connection.

But: the real issue is that Cyberduck is reporting success when the transfer is, in fact, an empty file and it won't resolve this issue until it is relaunched.

If there's no more discussion I'm going to report this as a bug.

**Reply by dkocher** [reply]
Please attach the transcript [^1] of a failed transfer.

[^1]: https://docs.cyberduck.io/cyberduck/support/#logging-output

**Reply by chrispink** [reply]
the file transferred (with 0 B result) was blocks.php.

**Reply by chrispink** [reply]
[Transfer log.txt](https://github.com/iterate-ch/cyberduck/files/12216281/Transfer.log.txt)

**Reply by chrispink** [reply]
Just to add, I have verified by other means that the file 'transferred' is truly empty.

**Reply by dkocher** [reply]
The server indicates the file is transferred completely. 

```
2023-07-31 15:44:05.927641+0100 0x5136e    Info        0x0                  17156  0    Cyberduck: (libcore.dylib) [ch.sudo.cyberduck:transcript] STOR /public_html/renovotec.dev/wordpress/wp-content/themes/renovotec/template-parts/blocks.php
2023-07-31 15:44:05.998602+0100 0x5136e    Info        0x0                  17156  0    Cyberduck: (libcore.dylib) [ch.sudo.cyberduck:transcript] 150 Accepted data connection
2023-07-31 15:44:06.066945+0100 0x5136e    Info        0x0                  17156  0    Cyberduck: (libcore.dylib) [ch.sudo.cyberduck:transcript] 226 File successfully transferred
```

Does it depend if the file already exists on the server and is replaced?

**Reply by chrispink** [reply]
Not sure. I will try. 

Once one file has failed, no other will succeed. 

I have already stated that the transfer window reports success. But the file (viewed through cPanel) is empty.

**Reply by chrispink** [reply]
Same result. A text file created locally and uploaded reports success in transcript and 0 B on the server.

**Reply by dkocher** [reply]
Looks like an issue with the server.

**Reply by chrispink** [reply]
<img width="837" alt="Screenshot 2023-07-31 at 16 30 49" src="https://github.com/iterate-ch/cyberduck/assets/9802285/d140a6b1-ba4d-4126-a949-b3e86f0a551a">

**Reply by chrispink** [reply]
![Screenshot 2023-07-31 at 16 29 03](https://github.com/iterate-ch/cyberduck/assets/9802285/b6347487-caeb-4b55-baa0-d4f2c5dbaaf1)

**Reply by chrispink** [reply]
![Screenshot 2023-07-31 at 16 28 45](https://github.com/iterate-ch/cyberduck/assets/9802285/86de2013-2098-4816-a25a-83ad13adf172)

**Reply by chrispink** [reply]
![Screenshot 2023-07-31 at 16 29 37](https://github.com/iterate-ch/cyberduck/assets/9802285/494fb292-99d3-4eca-8e85-cdad89af3abb)

**Reply by chrispink** [reply]
I have two windows open, Cyberduck and FileZilla. I can transfer with Cyberduck and get a 0 B on the server, then transfer the same file with FileZilla and get the expected result (repeatedly and consistently).

So, if it is the server, I'd appreciate a clue as what the difference is that is making the server behave differently for the two FTP programmes.

I will say that I don't this is IP switching as I've monitored my IP during this test and it has remained constant.

**Reply by chrispink** [reply]
And, if it is the server, why does relaunching Cyberduck reset the issue (briefly)?
